src/main.rs

```diff
@@ -0,0 +1,439 @@
+) -> Result<(), Box<dyn std::error::Error>> {
+    info!("Deploying all profiles for `{}`", node_name);
+
+    let mut profiles_list: Vec<&str> = node
+        .node_settings
+        .profiles_order
+        .iter()
+        .map(|x| x.as_ref())
+        .collect();
+
+    // Add any profiles which weren't in the provided order list
+    for profile_name in node.node_settings.profiles.keys() {
+        if !profiles_list.contains(&profile_name.as_str()) {
+            profiles_list.push(&profile_name);
+        }
+    }
+
+    for profile_name in profiles_list {
+        let profile = match node.node_settings.profiles.get(profile_name) {
+            Some(x) => x,
+            None => good_panic!("No profile was found named `{}`", profile_name),
+        };
+
+        let mut merged_settings = top_settings.clone();
+        merged_settings.merge(node.generic_settings.clone());
+        merged_settings.merge(profile.generic_settings.clone());
+
+        let deploy_data = utils::make_deploy_data(
+            top_settings,
+            node,
+            node_name,
+            profile,
+            profile_name,
+            cmd_overrides,
+        )?;
+
+        let deploy_defs = deploy_data.defs();
+
+        utils::deploy::deploy_profile(&deploy_data, &deploy_defs).await?;
+    }
+
+    Ok(())
+}
+
+/// Returns if the available Nix installation supports flakes
+#[inline]
+async fn test_flake_support() -> Result<bool, Box<dyn std::error::Error>> {
+    debug!("Checking for flake support");
+
+    Ok(Command::new("nix")
+        .arg("eval")
+        .arg("--expr")
+        .arg("builtins.getFlake")
+        .stdout(Stdio::null())
+        .stderr(Stdio::null())
+        .status()
+        .await?
+        .success())
+}
+
+async fn check_deployment(supports_flakes: bool, repo: &str, extra_build_args: &[String]) -> () {
+    info!("Running checks for flake in {}", repo);
+
+    let mut c = match supports_flakes {
+        true => Command::new("nix"),
+        false => Command::new("nix-build"),
+    };
+
+    let mut check_command = match supports_flakes {
+        true => {
+            c.arg("flake")
+                .arg("check")
+                .arg(repo)
+        }
+        false => {
+            c.arg("-E")
+                .arg("--no-out-link")
+                .arg(format!("let r = import {}/.; in (if builtins.isFunction r then (r {{}}) else r).checks.${{builtins.currentSystem}}", repo))
+        }
+    };
+
+    for extra_arg in extra_build_args {
+        check_command = check_command.arg(extra_arg);
+    }
+
+    let check_status = match check_command.status().await {
+        Ok(x) => x,
+        Err(err) => good_panic!("Error running checks for the given flake repo: {:?}", err),
+    };
+
+    if !check_status.success() {
+        good_panic!("Checks failed for the given flake repo");
+    }
+
+    ()
+}
+
+/// Evaluates the Nix in the given `repo` and return the processed Data from it
+async fn get_deployment_data(
```

Comment:
Maybe consider only partially evaluating the `deploy`, so that we don't have to wait any longer than we have to

Reply:
The evaluation time is certainly annoying but I think it's time I take a break from making so many major changes, if somebody else could do it that would be ideal